git/init.go

```diff
@@ -111,5 +120,42 @@ func Init(c *Client, opts InitOptions, dir string) (*Client, error) {
 	}
 	c.WorkDir = WorkDir(wd)
 	c.GitDir = GitDir(wd + "/.git")
+
+	if opts.Template != "" {
+		err = filepath.Walk(opts.Template, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+                        path, err = filepath.Rel(opts.Template, path)
+                        if err != nil {
+                                return err
+                        }
+			if info.IsDir() {
+				os.MkdirAll(filepath.Join(c.GitDir.String(), path), 0770)
+			} else {
+				newFile, err := os.Create(filepath.Join(c.GitDir.String(), path))
+				if err != nil {
+					return err
+				}
+				defer newFile.Close()
+				f, err := os.Open(filepath.Join(opts.Template, path))
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+				_, err = io.Copy(newFile, f)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			return c, err
+		}
```

Comment:
I don't think this needs the `err != nil` check. If err is nil, it falls through to the exact same return (and it doesn't look like code that should fall through if anything is ever added below it.)

Reply:
Thanks, I think I have addressed your comments with the latest commits. Can you have another look?